### DISCUSSION TITLE
how to fix "Invalid regular expression: /( |,|\?|!|;/: Unterminated group"

### DISCUSSION THREAD

**Discussion by dieshangtianya** [i18next/i18next#1742] [Q&A]
I introduce i18next and react-i18next, in most devices the app works great, but in some other android pad,  I encounter a bug that issued by sentry: “Invalid regular expression: /( |,|\?|!|;/: Unterminated group“，I dug into the sourcemap and found it occurred in the **looksLikeObjectPath** function. 

```
var chars = [' ', ',', '?', '!', ';'];
function looksLikeObjectPath(key, nsSeparator, keySeparator) {
  nsSeparator = nsSeparator || '';
  keySeparator = keySeparator || '';
  var possibleChars = chars.filter(function (c) {
    return nsSeparator.indexOf(c) < 0 && keySeparator.indexOf(c) < 0;
  });
  if (possibleChars.length === 0) return true;
  var r = new RegExp("(".concat(possibleChars.map(function (c) {
    return c === '?' ? '\\?' : c;
  }).join('|'), ")"));
  var matched = !r.test(key);

  if (!matched) {
    var ki = key.indexOf(keySeparator);

    if (ki > 0 && !r.test(key.substring(0, ki))) {
      matched = true;
    }
  }

  return matched;
}

```

I am wonder what happen in these devices! why does it happen only in some times? it is only a normal regular expression, and the nsSeparator, keySeparator variables are built-in.  I douted the string.concat function has been modified, maybe the nsSeparator, keySeparator has been changed in some other condition, but all assumptions are being overturned.

Anyone can give some advices about how to figure it out?

**Comment by adrai** [comment]
https://stackoverflow.com/questions/71523052/i18next-function-t-cause-error-invalid-regular-expression-unter

**Reply by yanghuabei** [reply]
The link is 404, please check

**Reply by adrai** [reply]
seems stackoverflow did delete that...
@yanghuabei please provide a minimal reproducible example and exact instructions if you are seeing that error
